### PR TITLE
i-bem: setMod/hasMod cast non-boolean mod values to string

### PR DESCRIPTION
**Pull request by narqo**
port #893 onto v3
part of #890

/cc @dfilatov

### CODE REVIEW

File: common.blocks/i-bem/i-bem.spec.js [hunk 1]
```diff
@@ -187,14 +193,14 @@ describe('i-bem', function() {
                 block.hasMod('mod1').should.be.true;
             });
 
-            it('in short form should return true for empty mod\'s value', function() {
+            it('in short form should return false for empty mod\'s value', function() {
```
Comment (narqo) [inline common.blocks/i-bem/i-bem.spec.js:196]: I'm not sure about why there was "true" if `shoud.be.false` is checked inside the test?

File: common.blocks/i-bem/i-bem.vanilla.js [hunk 2]
```diff
@@ -257,12 +257,15 @@ var BemEntity = inherit(events.Emitter, /** @lends BemEntity.prototype */ {
     /**
      * Checks whether a BEM entity has a modifier
      * @param {String} modName Modifier name
-     * @param {String} [modVal] Modifier value
+     * @param {String|Boolean} [modVal] Modifier value. If not of type String or Boolean, it is casted to String
      * @returns {Boolean}
      */
     hasMod : function(modName, modVal) {
+        var withModVal = arguments.length === 2;
```
Comment (narqo) [inline common.blocks/i-bem/i-bem.vanilla.js]: Can we decide that `hasMod('mod1', undefined)` is a useless edge case? At least until someone will catch this.

Comment (aristov) [inline common.blocks/i-bem/i-bem.vanilla.js]: No, it's important.

**Comment by narqo** [comment]
@dfilatov ping!

**Comment by dfilatov** [comment]
lgtm

File: common.blocks/i-bem/i-bem.spec.js (hunk 1 above)
Comment (qfox) [inline common.blocks/i-bem/i-bem.spec.js:196]: Looks like copy-paste issues.

**Pull request merged**